packages/envconfig/src/envconfig-toml.ts

```diff
@@ -0,0 +1,326 @@
+        );
+      }
+    }
+  }
+}
+
+function getFallbackConfigData(envProvider: EnvProvider): string | undefined {
+  // configSource was not set - fallback to TEMPORAL_CONFIG_FILE, then the default file path
+  let filePath = getEnvVar(envProvider, 'TEMPORAL_CONFIG_FILE');
+  if (filePath === undefined) {
+    filePath = getDefaultConfigFilePath();
+  }
+  return readFileSync(filePath).toString();
+}
+
+export function tomlLoadClientConfigProfile(options: LoadClientProfileOptions): TomlClientConfigProfile {
+  if (options.disableEnv && options.disableFile) {
+    throw new Error('Cannot disable both file and environment loading');
+  }
+
+  const envProvider: EnvProvider = options.overrideEnvVars
+    ? { kind: 'Map', map: options.overrideEnvVars }
+    : { kind: 'System' };
+
+  let profile: TomlClientConfigProfile = {};
+
+  if (!options.disableFile) {
+    const tomlClientConfig = tomlLoadClientConfig({
+      configSource: options.configSource,
+      configFileStrict: options.configFileStrict,
+      overrideEnvVars: options.overrideEnvVars,
+    });
+    // If profile name not provided, fallback to env variable.
+    const profileName = options.profile ?? getEnvVar(envProvider, 'TEMPORAL_PROFILE');
+    // If env var also not provided, fallback to default profile name.
+    const tomlProfile = tomlClientConfig.profile[profileName ?? DEFAULT_CONFIG_FILE_PROFILE];
+    // If toml profile does not exist and an explicit profile was requested, error.
+    if (tomlProfile === undefined && profileName) {
+      throw new Error(`Profile '${profileName}' not found in config data`);
+    }
+    // Use loaded profile if exists, otherwise fallback to default profile.
+    profile = tomlProfile ?? {};
+  }
+
+  if (!options.disableEnv) {
+    applyProfileEnvVars(profile, envProvider);
+  }
+  return profile;
+}
+
+function applyProfileEnvVars(profile: TomlClientConfigProfile, envProvider: EnvProvider) {
+  profile.address = getEnvVar(envProvider, 'TEMPORAL_ADDRESS') ?? profile.address;
+  profile.namespace = getEnvVar(envProvider, 'TEMPORAL_NAMESPACE') ?? profile.namespace;
+  profile.api_key = getEnvVar(envProvider, 'TEMPORAL_API_KEY') ?? profile.api_key;
+  const tlsFromEnv = getTLSFromEnvVars(envProvider);
+  profile.tls = profile.tls || tlsFromEnv ? { ...profile.tls, ...tlsFromEnv } : undefined;
+  const codecFromEnv = getCodecFromEnvVars(envProvider);
+  profile.codec = profile.codec || codecFromEnv ? { ...profile.codec, ...codecFromEnv } : undefined;
+  applyGrpcMetaFromEnvVars(profile, envProvider);
+}
+
+function getTLSFromEnvVars(envProvider: EnvProvider): TomlClientConfigTLS | undefined {
+  const tlsConfig: TomlClientConfigTLS = {};
+  const tlsEnvVar = getEnvVar(envProvider, 'TEMPORAL_TLS');
+  if (tlsEnvVar !== undefined) {
+    tlsConfig.disabled = envVarToBool(tlsEnvVar);
+  }
+  const clientCertPath = getEnvVar(envProvider, 'TEMPORAL_TLS_CLIENT_CERT_PATH');
+  if (clientCertPath !== undefined) {
```

Comment:
To make the code a bit simpler, why not instead unconditionally assign the properties, then use `filterNullAndUndefined` to get only keep fields that have values.